inc/Engine/Optimization/DynamicLists/AbstractDataManager.php

```diff
@@ -157,4 +157,13 @@ private function put_lists_to_file( string $content ): bool {
 	private function set_lists_cache( $content ) {
 		set_transient( $this->get_cache_transient_name(), $content, $this->cache_duration );
 	}
+
+	/**
+	 * Removes the lists cache
+	 *
+	 * @return void
+	 */
+	public function remove_lists_cache() {
+		delete_transient( $this->get_cache_transient_name() );
```

Comment:
```suggestion
		delete_transient( $this->get_cache_transient_name() );
		$lists_filepath = $this->get_json_filepath();

		if ( ! $this->filesystem->exists( $lists_filepath ) ) {
			return;
		}

		$this->filesystem->delete( $lists_filepath );
```

Seems like we need to remove the json file from the config also, to go and get the data from the json in the plugin file.
I tested that and it works.

Reply:
I pushed the code to let @hanna-meda validate from her side not to delay the process, feel free to adjust the code if needed.